app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  skip_before_action :authorize
+
+  def new
+  end
+
+  def create
+    user = User.find_by(name: params[:name])
+    if user&.authenticate(params[:password])
```

Comment:
Not sure if the book outlined using the safe navigation operator here, but I've seen people inside of Shopify say they don't allow it in their codebases before. Taking a quick look I found [this blog post](https://www.theguild.nl/5-reasons-not-to-use-safe-navigation-operators/) that has a few reasons why it could be a pain in the ass.

Without poking through it too much, to avoid the `&` operator, you could probably split out the conditional a bit more. Using a kind of short circuit like...

```suggestion
    if user.present? && user.authenticate(params[:password])
```

This is really kind of anecdotal though, so 100% open to chatting about it or looping in someone more experienced with it for thoughts/other ways of going about the same behaviour.

Another way would be to assign `user` by using `find_by!` which will raise `ActiveRecord::NoRecordFound` that you can then rescue and control the flow differently for.